### PR TITLE
fix: params setup for pagination support in api

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Border cases, right now we had some code and logic for calling paginated endpoints, but we were not setting this correctly on the request body, so we were stuck in a loop trying to get always more pages. 

This fixes the issue, creating a request class that can be used in paginated endpoints, right now: Conversations and Connections.

Thanks to @vitorhugods for the detailed eagle eye 🦅 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/pagination/PaginationRequest.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class PaginationRequest(
+    @SerialName("paging_state")
+    val pagingState: String?,
+    @SerialName("size")
+    val size: Int? = null // Set in case you want specific number of pages, otherwise, the backend will return default per endpoint
```
Comment (trOnk12) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/pagination/PaginationRequest.kt:11]: Perhaps worth to mention what the default size is ? I even wonder if we want to relay on backend default size?

**Review by trOnk12 — APPROVED** [review]
First step to pagination yey

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/pagination/PaginationRequest.kt (hunk 1 above)
Comment (yamilmedina) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/pagination/PaginationRequest.kt:11]: It depends, from endpoint to endpoint, for example connection requests default size is 100, for conversations 1000 😅 
So the idea is you can handle on the client/endpoint if you want something specific

Comment (trOnk12) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/pagination/PaginationRequest.kt:11]: Small details but what would be the case to relay on backend default values wouldnt it make more sense to have it set on the clients?

**Pull request merged**